File: QuevedoVirtualClassrooms-Server/src/main/java/com/quevedo/virtualclassroomsserver/common/models/dto/resource/ResourceDetailGetDTO.java
package com.quevedo.virtualclassroomsserver.common.models.dto.resource;

import com.quevedo.virtualclassroomsserver.common.models.common.ResourceType;
import com.quevedo.virtualclassroomsserver.common.models.server.resource.ResourceComment;
import com.quevedo.virtualclassroomsserver.common.models.server.user.UserVisualization;
import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
public class ResourceDetailGetDTO {
    private String uuidResource;
    private String resourceName;
    private String resourceEndpoint;
    private LocalDateTime timestamp;
    private String classroomUUID;
    private ResourceType resourceType;
    private List<ResourceComment> comments;
    private List<UserVisualization> visualizations;
}

File: QuevedoVirtualClassrooms-Android/app/src/main/java/org/quevedo/quevedovirtualclassrooms/data/models/resource/ResourceDetailGetDTO.kt
package org.quevedo.quevedovirtualclassrooms.data.models.resource

import org.quevedo.quevedovirtualclassrooms.data.models.common.ResourceType
import java.time.LocalDateTime

data class ResourceDetailGetDTO(
    val uuidResource: String,
    val resourceName: String,
    var resourceUrl: String,
    val timestamp: LocalDateTime,
    val classroomUUID: String,
    val resourceType: ResourceType,
    val comments: List<ResourceCommentDTO>
)